src/components/data_update/DatasetUpdate.js

```diff
@@ -0,0 +1,392 @@
+import { useCallback, useState, useEffect } from 'react';
+import { button, h, h2, div } from 'react-hyperscript-helpers';
+import { find, isArray, isNil } from 'lodash/fp';
+
+import { FormFieldTypes, FormField, FormValidators } from '../forms/forms';
+import { DAC, DAR, DataSet } from '../../libs/ajax';
+import { Notifications } from '../../libs/utils';
+
+export const DatasetUpdate = (props) => {
+  const { dataset } = props;
+
+  const [formData, setFormData] = useState({ dac: {}, dataUse: {}, properties: {} });
+
+  const searchOntologies = async (query, callback) => {
+    let options = [];
+    await DAR.getAutoCompleteOT(query).then(
+      items => {
+        options = items.map((item) => {
+          return item.label;
+        });
+        callback(options);
+      });
+  };
+
+  const dacOptions = (dacs) => {
+    let options = [];
+    if (!isNil(dacs)) {
+      options = dacs.map((dac) => {
+        return { displayText: dac.name, dacId: dac.dacId };
+      });
+    }
+    return options;
+  };
+
+  const getDiseaseLabels = async (ontologyIds) => {
+    let labels = [];
+    if (!isNil(ontologyIds)) {
+      const idList = ontologyIds.join(',');
+      const result = await DAR.searchOntologyIdList(idList);
+      labels = result.map(r => r.label);
+    }
+    return labels;
+  };
+
+  const extract = useCallback((propertyName) => {
+    const property = find({ propertyName })(dataset.properties);
+    return property?.propertyValue;
+  }, [dataset]);
+
+  const asProperty = (propertyName, propertyValue) => {
+    return {
+      propertyName,
+      propertyValue
+    };
+  };
+
+  const normalizeDataUse = useCallback(async (dataUse) => {
+    let du = dataUse;
+    if (!isNil(dataUse.diseaseRestrictions)) {
+      du.hasDiseaseRestrictions = true;
+      du.diseaseLabels = await getDiseaseLabels(dataUse.diseaseRestrictions);
+    }
+    if (!isNil(dataUse.other)) {
+      du.hasPrimaryOther = true;
+    }
+    if (!isNil(dataUse.secondaryOther)) {
+      du.hasSecondaryOther = true;
+    }
+    return du;
+  }, []);
+
+  const submitForm = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const formElement = event.target.form;
+    const submitData = new FormData(formElement);
+    const nihFileData = submitData.get('nihInstitutionalCertificationFile');
+    const consentGroups = [{ nihInstitutionalCertificationFile: nihFileData }];
+
+    const newDataset = {
+      name: dataset.name,
+      dacId: formData.dac.dacId,
+      properties: [
```

Comment:
Although the OpenAPI schema for DatasetUpdate says it requires a `datasetProperties` object, the `DatasetUpdate` object in Consent actually requires a `properties` object instead.

Reply:
Nice catch, we should file a bug fix for that.

Reply:
https://broadworkbench.atlassian.net/browse/DUOS-2579